modules/core/src/main/scala/doobie/util/write.scala

```diff
@@ -124,7 +124,8 @@ object Write extends LowerPriority1Write {
 
     override def toList(a: Option[A]): List[Any] = List(a)
 
-    override def toOpt: Write[Option[Option[A]]] = new Composite[Option[Option[A]]](List(this), x => List(x.flatten))
```

Comment:
Oops. These are actually fine as they are just creating an instance of `Composite`. 

e.g.
```
class Foo(i: Int)

new Foo(1) // fine, creates an instance of foo

new Foo(2) {} // anonymous subclass of Foo

new Foo(1) {
  def abc(): Unit = ...
}  // less fine, this is an anonymous subclass
```

Some instances I can find in the codebase where anonymous subclasses are created are `Get/Put/Query/Query0/Update/Update0`. 
`Get/Put` shouldn't be too hard to fix, I'd leave the others as changing them is a bit more tricky.

Reply:
Reverted